### PR TITLE
added initial single feature visualization mixin

### LINKED ISSUE #18: Single feature analysis tools (augmenting Rank1D)
Radviz, parallel coordinates, and Rank2D are already implemented. We should expand the feature visualization toolset to include single feature analysis tools:
- [ ] boxplots
- [ ] histograms
- [ ] violinplots

Create a `SingleFeatureVisualizer` class whose `poof` method takes as input the name(s) or position(s) of one of the features. Display the appropriate visualization allowing multiple `poof` calls with different features. Make sure to integrate this with Rank1D.

Open questions: do we need to store the data? Or can we create all visualization and then display on demand, or is there some middle ground? 

Note: similar to parallel coordinates and radviz, these need to be native implementations, emulating other libs (seaborn, pandas) as necessary.

### PR DESCRIPTION
**Pull request by ChrisPappalardo**
Fixes #18 

Single feature visualization mixin for arbitrary visualization classes that enables:

-  boxplots
-  histograms
-  violinplots

To use this mixin, simply create a new class that inherits both the visualizer you want to work with and the SingleFeatureVisualizationMixin, like so:

```
class SomeVizWithSFV(SingleFeatureVisualizerMixin, SomeViz):
    pass
```

Continue with your workflow as usual, replacing SomeViz with SomeVizWithSFV in your code.  Then if you want a visualize a feature along with whatever SomeViz produces, call poof with plotsingle set to one of 'box', 'hist', or 'violin' and optionally set feature to the appropriate 0index of the feature you are interested in (otherwise the first feature will be selected).

Want to customize the single feature visualization plots?  No problem.  Just define one or all of boxplot_params, hist_params, and/or violinplot_params (dicts) on the class when you define it, like so:

```
class SomeVizWithSFV(SingleFeatureVisualizerMixin, SomeViz):
    hist_params = {
        'facecolor': 'b',
    }
```

A full example is here:
```
import pandas as pd
from yellowbrick.features.base import SingleFeatureVisualizerMixin
from yellowbrick.features.radviz import RadViz

class RadVizSFV(SingleFeatureVisualizerMixin, RadViz):
    pass

data = pd.read_csv('data/occupancy/occupancy.csv')
features = ["temperature", "relative humidity", "light", "C02", "humidity"]
classes = ['unoccupied', 'occupied']

# Extract the numpy arrays from the data frame
X = data[features].as_matrix()
y = data.occupancy.as_matrix()

# Instantiate the visualizer
visualizer = RadVizSFV(classes=classes, features=features)

# fit and transform
visualizer.fit(X, y)
visualizer.transform(X)

# this will produce the RadViz visualization plus a histogram of the first feature
visualizer.poof(plotsingle='hist')
```

@DistrictDataLabs/team-oz-maintainers

### CODE REVIEW

**Comment by bbengfort** [comment]
@ChrisPappalardo thank you so much for the PR; we're a little backed up thanks to all the contributions from PyCon - but we'll get to reviewing this soon!

**Comment by ChrisPappalardo** [comment]
@bbengfort no problem, happy to contribute.  I can squash the commits or make whatever changes the team feels appropriate.  thanks for hosting us at pycon, I enjoyed meeting and working with you all.

File: yellowbrick/features/base.py [hunk 1]
```diff
@@ -11,14 +11,15 @@
 # ID: base.py [2e898a6] benjamin@bengfort.com $
 
 """
-Base classes for feature visualizers and feature selection tools.
+Base classes and mixins for feature visualizers and feature selection tools.
```
[the remark this answers is not in the document]
Comment (ChrisPappalardo) [reply yellowbrick/features/base.py]: No problem.  I updated them where I noticed they were impacted by my changes, but I didn't do a cold read of the comments.

File: yellowbrick/features/base.py [hunk 2]
```diff
@@ -225,3 +226,77 @@ def fit(self, X, y=None, **kwargs):
 
         # Fit always returns self.
         return self
+
+##########################################################################
+## Data Visualizer Mixins
+##########################################################################
+
+class SingleFeatureVisualizerMixin(object):
+    '''
```
[the remark this answers is not in the document]
Comment (ChrisPappalardo) [reply yellowbrick/features/base.py]: Ok, I'll change those.

File: yellowbrick/features/base.py [hunk 3]
```diff
@@ -225,3 +226,77 @@ def fit(self, X, y=None, **kwargs):
...
+    hist_params = {
+        'bins': 50,
+        'normed': 1,
+        'facecolor': 'g',
+        'alpha': 0.75,
+    }
+
+    violinplot_params = {}
+
+    def fit(self, X, *args, **kwargs):
```
[the remark this answers is not in the document]
Comment (ChrisPappalardo) [reply yellowbrick/features/base.py]: That make sense.  I used fit to get access to the X variable that was passed as part of an existing method signature, but I see your proposed changes below handle that.

File: yellowbrick/features/base.py [hunk 4]
```diff
@@ -225,3 +226,77 @@ def fit(self, X, y=None, **kwargs):
...
+        '''
+        store X from fit method on object for later use and call parent
+        '''
+
+        self.X = X
+
+        super(SingleFeatureVisualizerMixin, self).fit(X, *args, **kwargs)
+
+    def poof(self, plotsingle=None, feature=None, *args, **kwargs):
```
[the remark this answers is not in the document]
Comment (ChrisPappalardo) [reply yellowbrick/features/base.py]: This all makes sense to me (I'm fairly new to matplotlib and am still a bit confused by the implementation).  I'll work on your changes this week and push to the PR for your review.  Sorry for the late responses, I was on vacation for the past two weeks following pycon, just returned last night.

[the remark this answers is not in the document]
Comment (bbengfort) [reply yellowbrick/features/base.py]: No worries at all, thanks for contributing! Looking forward to seeing the new changes!

**Comment by rebeccabilbro** [comment]
Hey there @ChrisPappalardo - since the conversation has gone a bit stale here, I'm going to go ahead and close this one. Please feel free to open up a new PR as soon as you're ready to get back into this. We'd love to have you contribute!

**Pull request closed without merging**